pkg/injector/webhook.go

```diff
@@ -437,7 +443,79 @@ func updateMutatingWebhookCABundle(cert certificate.Certificater, webhookName st
 	return nil
 }
 
-func webhookExists(mwc admissionRegistrationTypes.MutatingWebhookConfigurationInterface, webhookName string) error {
-	_, err := mwc.Get(context.Background(), webhookName, metav1.GetOptions{})
-	return err
+func createMutatingWebhook(clientSet kubernetes.Interface, cert certificate.Certificater, webhookTimeout int32, webhookName, meshName, osmNamespace, osmVersion string) error {
+	webhookPath := webhookCreatePod
+	webhookPort := int32(constants.InjectorWebhookPort)
+	failuerPolicy := admissionregv1.Fail
+	matchPolict := admissionregv1.Exact
+
+	mwhc := admissionregv1.MutatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: webhookName,
+			Labels: map[string]string{
+				constants.OSMAppNameLabelKey:     constants.OSMAppNameLabelValue,
+				constants.OSMAppInstanceLabelKey: meshName,
+				constants.OSMAppVersionLabelKey:  osmVersion,
+				"app":                            constants.OSMInjectorName,
+				constants.ReconcileLabel:         strconv.FormatBool(true)}},
+		Webhooks: []admissionregv1.MutatingWebhook{
+			{
+				Name: MutatingWebhookName,
+				ClientConfig: admissionregv1.WebhookClientConfig{
+					Service: &admissionregv1.ServiceReference{
+						Namespace: osmNamespace,
+						Name:      constants.OSMInjectorName,
+						Path:      &webhookPath,
+						Port:      &webhookPort,
+					},
+					CABundle: cert.GetCertificateChain()},
+				FailurePolicy: &failuerPolicy,
+				MatchPolicy:   &matchPolict,
+				NamespaceSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						constants.OSMKubeResourceMonitorAnnotation: meshName,
+					},
+					MatchExpressions: []metav1.LabelSelectorRequirement{
+						{
+							Key:      constants.IgnoreLabel,
+							Operator: metav1.LabelSelectorOpDoesNotExist,
+						},
+						{
+							Key:      "name",
+							Operator: metav1.LabelSelectorOpNotIn,
+							Values:   []string{osmNamespace},
+						},
+						{
+							Key:      "control-plane",
+							Operator: metav1.LabelSelectorOpDoesNotExist,
+						},
+					},
+				},
+				Rules: []admissionregv1.RuleWithOperations{
+					{
+						Operations: []admissionregv1.OperationType{admissionregv1.Create},
+						Rule: admissionregv1.Rule{
+							APIGroups:   []string{"*"},
```

Comment:
Should this be `""` or `"*"`? If `""` works from the Go code, let's use that for consistency with the chart value.

Reply:
`""` doesn't work in go code, it will consider it as empty and omit it. which causes the creation of the webhook to fail since its a mandatory field. Hence it has to be `"*"`. Do we want to change the chart to `"*"` ?

Reply:
I think this is fine. Thanks for clarifying!